src/Sylius/Bundle/CoreBundle/Validator/Constraints/ChannelCodeCollectionValidator.php

```diff
@@ -38,10 +43,44 @@ public function validate(mixed $value, Constraint $constraint): void
             throw new UnexpectedValueException($value, 'array');
         }
 
+        if ($constraint->validateAgainstAllChannels) {
+            $this->validateInChannelCollection($value, $this->channelRepository->findAll(), $constraint);
+
+            return;
+        }
+
+        $object = $this->context->getObject();
+
+        if (null !== $constraint->channelAwarePropertyPath) {
+            $object = $this->propertyAccessor->getValue($object, $constraint->channelAwarePropertyPath);
+        }
+
+        if (!$object instanceof ChannelsAwareInterface) {
+            throw new \LogicException(sprintf(
+                'The validated root needs to implement the %s interface when option `validateAgainstAllChannels` is set to false.',
```

Comment:
```suggestion
                'The validated object needs to implement the %s interface when option `validateAgainstAllChannels` is set to false.',
```
Since it's not the root anymore